tests/api2/pool_dataset_encryption.py

```diff
@@ -0,0 +1,870 @@
+
+
+@skip_for_ha
+def test_28_delete_not_encrypted_dataset():
+    results = DELETE(f'/pool/dataset/id/{dataset_url}/')
+    assert results.status_code == 200, results.text
+
+
+@skip_for_ha
+def test_29_create_a_dataset_to_inherit_encryption_from_the_passphrase_encrypted_pool():
+    payload = {
+        'name': dataset,
+        'inherit_encryption': True
+    }
+    results = POST('/pool/dataset/', payload)
+    assert results.status_code == 200, results.text
+
+
+@skip_for_ha
+def test_30_delete_encrypted_dataset():
+    results = DELETE(f'/pool/dataset/id/{dataset_url}/')
+    assert results.status_code == 200, results.text
+
+
+@skip_for_ha
+def test_31_try_to_create_an_encrypted_root_with_generate_key_on_passphrase_encrypted_pool():
+    payload = {
+        'name': dataset,
+        'encryption_options': {
+            'generate_key': True,
+        },
+        'encryption': True,
+        'inherit_encryption': False
+    }
+    results = POST('/pool/dataset/', payload)
+    assert results.status_code == 422, results.text
+
+
+@skip_for_ha
+def test_32_try_to_create_an_encrypted_root_with_key_on_passphrase_encrypted_pool():
+    payload = {
+        'name': dataset,
+        'encryption_options': {
+            'key': dataset_token_hex,
+        },
+        'encryption': True,
+        'inherit_encryption': False
+    }
+    results = POST('/pool/dataset/', payload)
+    assert results.status_code == 422, results.text
+
+
+@skip_for_ha
+def test_33_delete_the_passphrase_encrypted_pool_with_is_datasets():
+    payload = {
+        'cascade': True,
+        'restart_services': True,
+        'destroy': True
+    }
+    results = POST(f'/pool/id/{pool_id}/export/', payload)
+    assert results.status_code == 200, results.text
+    job_id = results.json()
+    job_status = wait_on_job(job_id, 120)
+    assert job_status['state'] == 'SUCCESS', str(job_status['results'])
+
+
+@skip_for_ha
+def test_34_creating_a_key_encrypted_pool():
+    global pool_id
+    payload = {
+        'name': pool_name,
+        'encryption': True,
+        'encryption_options': {
+            'algorithm': 'AES-128-CCM',
+            'key': pool_token_hex,
+        },
+        'topology': {
+            'data': [
+                {'type': 'STRIPE', 'disks': disk_pool}
+            ],
+        }
+    }
+    results = POST('/pool/', payload)
+    assert results.status_code == 200, results.text
+    job_id = results.json()
+    job_status = wait_on_job(job_id, 120)
+    assert job_status['state'] == 'SUCCESS', str(job_status['results'])
+    pool_id = job_status['results']['result']['id']
+
+
+@skip_for_ha
+def test_35_creating_a_key_encrypted_root_on_key_encrypted_pool():
+    payload = {
+        'name': dataset,
+        'encryption_options': {
+            'key': dataset_token_hex,
+        },
+        'encryption': True,
+        'inherit_encryption': False
+    }
+    results = POST('/pool/dataset/', payload)
+    assert results.status_code == 200, results.text
```

Comment:
Please assert that the dataset is indeed encrypted with a `HEX` key and that it is indeed an encryption root itself